apis/database/v1alpha1/types.go

```diff
@@ -0,0 +1,203 @@
+/*
+Copyright 2019 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	runtimev1alpha1 "github.com/crossplaneio/crossplane-runtime/apis/core/v1alpha1"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/crossplaneio/stack-rook/apis/v1alpha1"
+)
+
+// ServerSpec describes server related settings of the cluster
+type ServerSpec struct {
```

Comment:
I do still have a **slight** concern about redeclaring the Rook CRDs here within this stack, but I agree with the points from @negz and @hasheddan:

* It's good practice to be in control of the public API you expose, don't directly expose code outside of your control into the public API that you maintain.
* This is an analogous approach that we have taken with the cloud provider APIs
* This is the first implementation and we can assess the maintenance burden along the way